src/net/neighbors/mod.rs

```diff
@@ -0,0 +1,430 @@
+pub const WALK_RESET_INTERVAL: u64 = 600;
+
+/// How often the node will consider pruning neighbors from its neighbor set.  The node will prune
+/// neighbors from over-represented hosts and IP ranges in order to maintain connections to a
+/// diverse set of neighbors.
+#[cfg(test)]
+pub const PRUNE_FREQUENCY: u64 = 0;
+#[cfg(not(test))]
+pub const PRUNE_FREQUENCY: u64 = 43200;
+
+/// Not all neighbors discovered will have an up-to-date chain tip.  This value is the highest
+/// discrepancy between the local burnchain block height and the remote node's burnchain block
+/// height for which the neighbor will be considered as a worthwhile peer to remember.
+#[cfg(test)]
+pub const MAX_NEIGHBOR_BLOCK_DELAY: u64 = 25;
+#[cfg(not(test))]
+pub const MAX_NEIGHBOR_BLOCK_DELAY: u64 = 288;
+
+/// How often to kick off neighbor walks.
+#[cfg(test)]
+pub const NEIGHBOR_WALK_INTERVAL: u64 = 0;
+#[cfg(not(test))]
+pub const NEIGHBOR_WALK_INTERVAL: u64 = 120; // seconds
+
+impl PeerNetwork {
+    /// Instantiate a neighbor walk, and update internal bookkeeping about how many times and how
+    /// often we've done this (so we can intelligently alter walk strategies).
+    ///
+    /// Returns the new neighbor walk on success.
+    /// Returns None if we could not instantiate a walk for some reason.
+    fn new_neighbor_walk(
+        &mut self,
+        ibd: bool,
+    ) -> Option<NeighborWalk<PeerDBNeighborWalk, PeerNetworkComms>> {
+        // alternate between starting walks from inbound and outbound neighbors.
+        // fall back to pingbacks-only walks if no options exist.
+        debug!(
+            "{:?}: Begin walk attempt {}",
+            &self.local_peer, self.walk_attempts
+        );
+
+        // always ensure we're connected to always-allowed outbound peers
+        let walk_res = if ibd {
+            // always connect to bootstrap peers if in IBD
+            NeighborWalk::instantiate_walk_to_always_allowed(
+                self.get_neighbor_walk_db(),
+                self.get_neighbor_comms(),
+                self,
+                ibd,
+            )
+        } else {
+            // if not in IBD, then we're not required to use the always-allowed neighbors
+            // all the time (since they may be offline, and we have all the blocks anyway).
+            // Alternate between picking random neighbors, and picking always-allowed
+            // neighbors.
+            if self.walk_attempts % (self.connection_opts.walk_inbound_ratio + 1) == 0 {
+                NeighborWalk::instantiate_walk(
+                    self.get_neighbor_walk_db(),
+                    self.get_neighbor_comms(),
+                    self,
+                )
+            } else {
+                NeighborWalk::instantiate_walk_to_always_allowed(
+                    self.get_neighbor_walk_db(),
+                    self.get_neighbor_comms(),
+                    self,
+                    ibd,
+                )
+            }
+        };
```

Comment:
```suggestion
        // always connect to bootstrap peers if in IBD. if not, alternate between random peers
        // and always connected ones 
        let walk_to_always_connected_peer = ibd || (self.walk_attempts % (self.connection_opts.walk_inbound_ratio + 1) != 0);
        let walk_res = if walk_to_always_connected_peer {
            NeighborWalk::instantiate_walk_to_always_allowed(
                self.get_neighbor_walk_db(),
                self.get_neighbor_comms(),
                self,
                ibd,
            )
        } else {
            NeighborWalk::instantiate_walk(
                    self.get_neighbor_walk_db(),
                    self.get_neighbor_comms(),
                    self,
                )
        };
```

Reply:
sure

Reply:
I'm going to tweak this slightly.  What I want to happen is if the node isn't IBD, then it should alternate between outbound walks, inbound walks, and pingback walks.